storage/localstore/mode_set.go

```diff
@@ -393,3 +399,25 @@ func (db *DB) setUnpin(batch *leveldb.Batch, addr chunk.Address) (err error) {
 
 	return nil
 }
+
+// setReUpload adds a pinned chunk to the push index so that it can be re-uploaded to the network
+func (db *DB) setReUpload(batch *leveldb.Batch, addr chunk.Address) (err error) {
+	item := addressToItem(addr)
+
+	// only pinned chunks should be re-uploaded
+	// this also prevents a race condition: a non-pinned chunk could be gc'd after it's put in the push index
+	// but before it is actually re-uploaded to the network
+	_, err = db.pinIndex.Get(item)
+	if err != nil {
+		return err
+	}
+
+	retrievalDataIndexItem, err := db.retrievalDataIndex.Get(item)
```

Comment:
`retrievalDataIndexItem` already contains all fields from `item`, so it can be passed to `PutInBatch` a few lines below without the need to explicitly set StoreTimestamp on `item`.

Reply:
thanks, removed the `StoreTimestamp` assignment and replaced:
```
db.pushIndex.PutInBatch(batch, item)
```
with:
```
db.pushIndex.PutInBatch(batch, retrievalDataIndexItem)
```